bluesky/plans.py

```diff
@@ -57,59 +57,18 @@ def count(detectors, num=1, delay=None, *, md=None):
     _md.update(md or {})
     _md['hints'].setdefault('dimensions', [(('time',), 'primary')])
 
-    # If delay is a scalar, repeat it forever. If it is an iterable, leave it.
-    if not isinstance(delay, Iterable):
-        delay = itertools.repeat(delay)
-    else:
-        try:
-            num_delays = len(delay)
-        except TypeError:
-            # No way to tell in advance if we have enough delays.
-            pass
-        else:
-            if num - 1 > num_delays:
-                raise ValueError("num=%r but delays only provides %r "
-                                 "entries" % (num, num_delays))
-        delay = iter(delay)
-
-    @bpp.stage_decorator(detectors)
-    @bpp.run_decorator(md=_md)
-    def finite_plan():
-        for i in range(num):
-            now = time.time()  # Intercept the flow in its earliest moment.
+    def outer():
+        def inner():
```

Comment:
It looks like we can just remove `inner` here. The outer/inner pattern is only needed if we need to decorate `inner` with some information derived from parameters passed to `outer`.